.golangci.yml

```diff
@@ -126,6 +126,9 @@ linters-settings:
       - pkg: sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1
         alias: controlplanev1
 
+  nolintlint:
+    # https://github.com/golangci/golangci-lint/issues/3228
+    allow-unused: true
```

Comment:
In this PR, it was problematic for the `nolint:godot` and I found the issue that explains this behaviour can be a known issue and the workaround for now is to allow unused nolints to avoid the error.